wfe/wfe.go

```diff
@@ -740,8 +742,22 @@ func (wfe *WebFrontEndImpl) RevokeCertificate(ctx context.Context, logEvent *req
 		return
 	}
 
-	// Use revocation code 0, meaning "unspecified"
-	err = wfe.RA.RevokeCertificateWithReg(ctx, *parsedCertificate, 0, registration.ID)
+	reason := core.RevocationCode(0)
+	if revokeRequest.Reason != nil && wfe.AcceptRevocationReason {
+		if _, present := core.RevocationReasons[*revokeRequest.Reason]; !present {
```

Comment:
Is it worth the extra if statement just to distinguish between invalid vs unsupported? I think we can always return "unsupported" and be happy with it.